### PR TITLE
Build and publish libuuid and zlib

### PR DESCRIPTION
**Pull request by assambar**
1. Modified Makefile-s and Dockerfile-s around builder containers. Now, running `make TARGET keep-containers=1` will only build missing containers and will reuse existing ones.
         
NOTE: Use this wisely if you change build scripts. As they get added to the containers. Respective source files in `libs`, `php`, `python`, `ruby` are always mounted on container run, so changes there are picked up with `keep-containers`.

 2. Add build scripts and GH actions to build and publish zlib-2.1.13. Modified python3.11.1 to use this zlib version. Sample release here - https://github.com/assambar/webassembly-language-runtimes/releases/tag/libs%2Fzlib%2F1.2.13%2B20230203-3b1b27c

 3. Add build scripts and GH actions to build and publish libuuid-1.0.3, instead of downloading it from singlestore-labs. A patch that mocks `umask` and `getuid` was necessary. Sample release here - https://github.com/assambar/webassembly-language-runtimes/releases/tag/libs%2Fuuid%2F1.0.3%2B20230203-236edf4

Notes:
------
 - Still pending checks after rebase on master
 - Have yet to decide on the exact archive structure and format of the published libs

### CODE REVIEW

File: .github/workflows/build-uuid.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    # running on tag push. We have a dedicated workflow to be ran when
+    # a tag is pushed.
+    branches:
+      - "*"
+    paths:
+       - .github/**
+       - 'libs/uuid/**'
+       - 'scripts/**'
+       - 'Makefile*'
+       - '*.sh'
```
Comment (ereslibre) [inline .github/workflows/build-uuid.yaml:14]: Super minor: unquoted vs quoted, to keep all of them the same.

Comment (ereslibre) [reply .github/workflows/build-uuid.yaml:14]: Applies to other workflows too, not reporting on all of them to reduce noise.

**Review by ereslibre — APPROVED** [review]
LGTM; thanks @assambar!

**Comment by gzurl** [comment]
LGTM!
Regarding the new `keep-containers=1` parameter, should that be the default behavior? I assume that most of the time we want to keep the containers. So I wonder about this: by default containers are reused, unless `rebuild-containers=1`. @assambar @ereslibre WDYT?

**Review by gzurl — APPROVED** [review]
👍 Just a suggestion regarding the new Make parameter.

**Comment by assambar** [comment]
> LGTM! Regarding the new `keep-containers=1` parameter, should that be the default behavior? I assume that most of the time we want to keep the containers. So I wonder about this: by default containers are reused, unless `rebuild-containers=1`. @assambar @ereslibre WDYT?

@gzurl what you suggest is exactly what I did initially. But then I had several cases of failures just because the containers had not picked up everything that changed. For example the "scripts" folder is built into the base container and not mounted explicitly.
Maybe we could add to what you are saying that we MUST NEVER include source from this repo in the containers. Instead we should always mount. And that applies for build scripts as well as source code. Then, we can skip rebuilding them by default.
Personally, I would like this approach, but do not have enough experience with such workflows to say it's a good way to go.

@ereslibre, do you also have some thoughts on this?

**Comment by ereslibre** [comment]
> @ereslibre, do you also have some thoughts on this?

Always mounting (both the projects and the build scripts) is fine. We haven't optimized for modifying the toplevel wlr scripts, but I would be fine mounting those as well.

From my POV it would be fine to go ahead with this change and create dedicated PR's to optimize the build environment so that we always mount and it's easier to modify any aspect of it (let it be the toplevel build scripts, project build scripts, or project staging source code).

**Comment by gzurl** [comment]
OK, let's do that folks! And let optimization for a different PR.

**Comment by ereslibre** [comment]
@assambar is anything blocking the rebase + merge? There are a couple of CI failures

**Comment by assambar** [comment]
Closes #29 and #30

**Pull request merged**